### PR TITLE
Use HDF5 for testing simplex workflows

### PR DESCRIPTION
**Pull request by elauksap**
Enable HDF5 in the simplex workflow.

### CODE REVIEW

**Review by peterrum — APPROVED** [review]

**Comment by elauksap** [comment]
This PR can be safely merged after #10886, that also fixes the `simplex/data_out_write_hdf5_01` test failing here.

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -62,6 +63,10 @@ jobs:
               -D DEAL_II_WITH_TRILINOS="ON" \
               -D DEAL_II_WITH_PETSC="ON" \
               -D DEAL_II_WITH_METIS="ON" \
+              -D DEAL_II_WITH_HDF5="ON" \
+              -D HDF5_INCLUDE_DIR="/usr/include/hdf5/openmpi/" \
+              -D HDF5_LIBRARY="/usr/lib/x86_64-linux-gnu/hdf5/openmpi/libhdf5.so" \
+              -D HDF5_HL_LIBRARY="/usr/lib/x86_64-linux-gnu/hdf5/openmpi/libhdf5_hl.so" \
```
Comment (tamiko) [inline .github/workflows/linux.yml:69]: It should suffice to simply specify something like `CMAKE_PREFIX_PATH="/usr/lib/x86_64-linux-gnu/hdf5/openmpi;/usr/include/hdf5/openmpi"`

**Review by tamiko — APPROVED** [review]

**Comment by peterrum** [comment]
@tamiko Why have you merged this PR? One test has not passed.

**Comment by elauksap** [comment]
@peterrum see the comment above, the failing test was fixed in #10886!

**Pull request merged**